E/Enzyme/build_tarballs.jl

```diff
@@ -137,6 +138,6 @@ for (i,build) in enumerate(builds)
     build_tarballs(i == lastindex(builds) ? non_platform_ARGS : non_reg_ARGS,
                    name, version, sources, script,
                    build.platforms, build.products, build.dependencies;
-                   preferred_gcc_version=v"8", julia_compat="1.6",
+                   preferred_gcc_version=build.gcc_version, julia_compat="1.6",
```

Comment:
I think this is still not working as expected, because host and target should have same version of GCC (unless there are platform-specific constraints):
```console
% julia --compile=min -e 'using BinaryBuilderBase; BinaryBuilderBase.runshell(Platform("x86_64", "linux"); preferred_gcc_version=v"10")'
sandbox:${WORKSPACE} # $CC --version
x86_64-linux-gnu-gcc (GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

sandbox:${WORKSPACE} # $HOSTCC --version
x86_64-linux-musl-gcc (GCC) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
If we ask for GCC 10, in general we should get GCC 10 for both target and host, but https://github.com/JuliaPackaging/Yggdrasil/pull/8017#issuecomment-1912657875 references GCC 8.

Reply:
Sounds reasonable to me, how does one force that?